public/app/plugins/datasource/prometheus/components/VariableQueryEditor.tsx

```diff
@@ -0,0 +1,249 @@
+      return;
+    }
+
+    labelNamesVarQuery(datasource).then((labelNames: Array<{ text: string }>) => {
+      setLabelOptions(labelNames.map(({ text }) => ({ label: text, value: text })));
+    });
+  }, [datasource, exprType]);
+
+  const onChangeWithVariableString = (exprType: number) => {
+    const queryVar = {
+      exprType: exprType,
+      label,
+      metric,
+      varQuery,
+      seriesQuery,
+      refId: 'PrometheusVariableQueryEditor-VariableQuery',
+    };
+
+    const queryString = migrateVariableEditorBackToVariableSupport(queryVar);
+
+    onChange({
+      query: queryString,
+      refId,
+    });
+  };
+
+  const onQueryTypeChange = (newType: SelectableValue<QueryType>) => {
+    setExprType(newType.value);
+    if (newType.value === QueryType.LabelNames) {
+      onChangeWithVariableString(newType.value);
+    }
+  };
+
+  const onLabelChange = (newLabel: SelectableValue<string>) => {
+    setLabel(newLabel.value ?? '');
+  };
+
+  const onMetricChange = (e: FormEvent<HTMLInputElement>) => {
+    setMetric(e.currentTarget.value);
+  };
+
+  const onVarQueryChange = (e: FormEvent<HTMLInputElement>) => {
+    setVarQuery(e.currentTarget.value);
+  };
+
+  const onSeriesQueryChange = (e: FormEvent<HTMLInputElement>) => {
+    setSeriesQuery(e.currentTarget.value);
+  };
+
+  const handleBlur = () => {
+    if (exprType === QueryType.LabelNames) {
+      onChangeWithVariableString(exprType);
+    } else if (exprType === QueryType.LabelValues && label) {
+      onChangeWithVariableString(exprType);
+    } else if (exprType === QueryType.MetricNames && metric) {
+      onChangeWithVariableString(exprType);
+    } else if (exprType === QueryType.VarQueryResult && varQuery) {
+      onChangeWithVariableString(exprType);
+    } else if (exprType === QueryType.SeriesQuery && seriesQuery) {
+      onChangeWithVariableString(exprType);
+    }
+  };
+
+  return (
+    <InlineFieldRow>
+      <InlineField
+        label="Function type"
+        labelWidth={20}
+        tooltip={<div>The Prometheus data source plugin provides the following functions for querying variables.</div>}
+      >
+        <Select
+          placeholder="Select function"
+          aria-label="Function type"
+          onChange={onQueryTypeChange}
+          onBlur={handleBlur}
+          value={exprType}
+          options={variableOptions}
+          width={25}
+        />
+      </InlineField>
+      {exprType === QueryType.LabelValues && (
+        <>
+          <InlineField label="Label" labelWidth={20}>
+            <Select
+              aria-label="label-select"
+              onChange={onLabelChange}
+              onBlur={handleBlur}
+              value={label}
+              options={labelOptions}
+              width={25}
+              allowCustomValue
+            />
+          </InlineField>
+          <InlineField
+            label="Metric"
+            labelWidth={20}
+            tooltip={
+              <div>
```

Comment:
I see this calling label values unless I'm using an old prom version (but both API endpoints appear to work correctly), so I'm not sure this is accurate, additionally I'm not sure the tooltips are an appropriate context for API level documentation, that should be available in the change notes, or the code, but this information presented to users should be oriented towards helping them understand and use this field.

Reply:
Got it! I'll remove the reference to the api.